packages/camera/camera_platform_interface/lib/src/types/media_settings.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: avoid_equals_and_hash_code_on_mutable_classes
+
+import 'resolution_preset.dart';
+
+/// Recording media settings.
+///
+/// Used in [CameraPlatform.createCameraWithSettings].
+/// Allows to tune recorded video parameters, such as resolution, frame rate, bitrate.
+/// If [fps], [videoBitrate] and [audioBitrate] passed, they must be greater than zero.
+class MediaSettings {
+  /// Creates a [MediaSettings].
+  const MediaSettings({
+    this.resolutionPreset,
+    this.fps,
+    this.videoBitrate,
+    this.audioBitrate,
+    this.enableAudio = false,
+  })  : assert(fps == null || fps > 0, 'fps must be null or greater than zero'),
+        assert(videoBitrate == null || videoBitrate > 0,
+            'videoBitrate must be null or greater than zero'),
+        assert(audioBitrate == null || audioBitrate > 0,
+            'audioBitrate must be null or greater than zero');
+
+  /// [ResolutionPreset] affect the quality of video recording and image capture.
+  final ResolutionPreset? resolutionPreset;
+
+  /// Rate at which frames should be captured by the camera in frames per second.
+  ///
+  /// Must be null or greater than zero
```

Comment:
You don't need to re-comment this on each member; since they are final it's impossible to set them other than via the constructor, and the constructor comments it now.

Reply:
removed excessive comments